New warnings in polynomial.cc

```
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc: In member function ‘void dealii::Polynomials::PiecewisePolynomial<number>::value(number, unsigned int, number*) const [with number = float]’:
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc:153:18:   instantiated from here
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc:61:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc: In member function ‘void dealii::Polynomials::PiecewisePolynomial<number>::value(number, unsigned int, number*) const [with number = double]’:
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc:154:18:   instantiated from here
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc:61:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc: In member function ‘void dealii::Polynomials::PiecewisePolynomial<number>::value(number, unsigned int, number*) const [with number = long double]’:
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc:155:18:   instantiated from here
/u/bangerth/p/deal.II/2/dealii/source/base/polynomials_piecewise.cc:61:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
[220/244] Building CXX object source/base/CMakeFiles/obj_base.debug.dir/polynomial.cc.o
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc: In member function ‘void dealii::Polynomials::Polynomial<number>::value(number, unsigned int, number*) const [with number = float]’:
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc:1371:18:   instantiated from here
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc:117:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc: In member function ‘void dealii::Polynomials::Polynomial<number>::value(number, unsigned int, number*) const [with number = double]’:
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc:1372:18:   instantiated from here
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc:117:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc: In member function ‘void dealii::Polynomials::Polynomial<number>::value(number, unsigned int, number*) const [with number = long double]’:
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc:1373:18:   instantiated from here
/u/bangerth/p/deal.II/2/dealii/source/base/polynomial.cc:117:5: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
```

@gassmoeller -- I suspect that is a fallout from #3138 .

--- a/source/base/polynomial.cc
+++ b/source/base/polynomial.cc
@@ -114,8 +114,6 @@ namespace Polynomials
                              const unsigned int n_derivatives,
                              number *values) const
   {
-    Assert(n_derivatives >= 0, ExcZero());
-
     // evaluate Lagrange polynomial and derivatives
     if (in_lagrange_product_form == true)
       {

--- a/source/base/polynomials_piecewise.cc
+++ b/source/base/polynomials_piecewise.cc
@@ -58,8 +58,6 @@ namespace Polynomials
                                       const unsigned int n_derivatives,
                                       number *values) const
   {
-    Assert (n_derivatives >= 0, ExcZero());
-
     // shift polynomial if necessary
     number y = x;
     double derivative_change_sign = 1.;

